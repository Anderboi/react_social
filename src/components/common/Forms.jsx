import base from "../../Common.module.css";

export const TextForm = ({ ...props }) => {
  return (
    <div>
      <input
        className={base.input}
        autoFocus
        value={props.value}
        onChange={props.onChange}
        name={props.name}
        id={props.id}
        placeholder={props.placeholder}
        validate={props.validate}
        onBlur={props.onBlur}
      ></input>
    </div>
  );
};
