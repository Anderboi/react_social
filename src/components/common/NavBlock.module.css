.nav_menu {
  display: flex;
  flex-direction: column;
  gap: 0.5em;
  align-items: flex-end;
  margin-bottom: 5em;
  
}

.nav_item {
  cursor: pointer;
  align-self: stretch;
  text-align: end;
  text-decoration: none;
}

.active {
  align-self: stretch;
  text-align: end;
  color: white;
  text-decoration: none;
  padding-right: 0.5em;
  border-bottom: 1px solid;
}

.friends_block {
  display: flex;
  flex-direction: row-reverse;
  align-items: center;
  gap: 0.4em;
  font-size: var(--fs-300);
}

.friends_block_item {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.friends_block_img {
  width: 40px;
}

@media screen and (max-width: 768px) {


}
