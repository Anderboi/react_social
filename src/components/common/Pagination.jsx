/* eslint-disable jsx-a11y/anchor-is-valid */
import css from "./Pagination.module.css";

const Pagination = ({
  selectedPage,
  pageSize,
  itemsTotalCount,
  onPageChanged,
  pagesRange
}) => {


  const pages = Math.ceil(itemsTotalCount / pageSize);

  const pageNumbers = [];

  for (let x = 1; x <= pages; x++) {
    pageNumbers.push(x);
  }

  return (
    <div className={`${css.pagination} ${css.page}`}>
      <div
        className={`${
          selectedPage < Math.ceil(pagesRange / 2) + 1 && css.page_hidden
        }`}
        onClick={() => {
          onPageChanged(1);
        }}
      >
        First
      </div>
      {pageNumbers
        .map((p) => {
          return (
            <a
              href="#"
              key={p}
              className={`${selectedPage === p && css.selectedPage} ${
                css.page
              }`}
              onClick={() => {
                onPageChanged(p);
              }}
            >
              {p}
            </a>
          );
        })
        .slice(
          selectedPage > Math.floor(pagesRange / 2) ? selectedPage - Math.ceil(pagesRange / 2) : 0,
          selectedPage > Math.floor(pagesRange / 2) ? selectedPage + Math.floor(pagesRange / 2) : pagesRange
        )}
      <div
        className={`${
          selectedPage > pages - Math.ceil(pagesRange / 2) && css.page_hidden
        }`}
        onClick={() => {
          onPageChanged(pages);
        }}
      >
        ...{pages}
      </div>
    </div>
  );
};

export default Pagination;
