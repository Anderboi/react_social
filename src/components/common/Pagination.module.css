.pagination {
  margin: 0 auto;
  width: 32em;
  display: grid;
  grid-template-columns: 3fr 2fr 2fr 2fr 2fr 2fr 2fr 2fr 3fr;
  grid-template-rows:  1fr;
}

.page {
  text-decoration: none;
  text-align: center;
  color: black;
  cursor: pointer;
}

.page_hidden {
  color: transparent;
  cursor: none;
}

.selectedPage {
  font-weight: bold;
  color: var(--main-color);
  border-bottom: 2px solid var(--main-color);
}
