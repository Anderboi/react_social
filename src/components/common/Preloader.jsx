import React from "react";
import css from "../../Common.module.css";
import spinner from "../../assets/images/Dual Ring-1s-200px.svg";

export const Preloader = () => {
  return (
    <div className={css.spinner}>
      <img src={spinner} alt="spinner" />
    </div>
  );
};
