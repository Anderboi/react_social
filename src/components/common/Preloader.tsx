import React from "react";
import css from "../../Common.module.css";
// @ts-ignore
import spinner from "../../assets/images/Dual Ring-1s-200px.svg";

export const Preloader = (): JSX.Element => {
  return (
    <div className={css.spinner}>
      <img src={spinner} alt="spinner" />
    </div>
  );
};
