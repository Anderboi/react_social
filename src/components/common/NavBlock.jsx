import React from "react";
import common from "../../Common.module.css";
import nav from "./NavBlock.module.css";
import { NavLink } from "react-router-dom";

const NavBlock = (props) => {
  const linkClass = (navData) =>
    `${navData.isActive ? nav.active : nav.nav_item} ${common.hover}`;

  return (
    <div
      className={`${
        !props.isForMobile ? nav.main_nav_block : nav.main_nav_block_mobile
      }`}
    >
      <nav className={nav.nav_menu}>
        <NavLink className={linkClass} to="/profile" data-testid="profile-link">
          Profile
        </NavLink>
        <NavLink
          className={linkClass}
          to="/messages"
          data-testid="messages-link"
        >
          Messages
        </NavLink>
        <NavLink className={linkClass} to="/news" data-testid="news-link">
          News
        </NavLink>
        <NavLink className={linkClass} to="/music" data-testid="music-link">
          Music
        </NavLink>
        <br></br>

        <NavLink className={linkClass} to="/Users" data-testid="users-link">
          Users
        </NavLink>
        <br></br>
        <NavLink
          className={linkClass}
          to="/settings"
          data-testid="settings-link"
        >
          Settings
        </NavLink>
        {/* <Outlet /> */}
      </nav>
   
      <div className={props.isAuth ? nav.nav_menu : common.display_none}>
        <h3>Friends</h3>
        <div className={nav.friends_block}>
          <FriendItem />
          <FriendItem />
          <FriendItem />
        </div>
      </div>
    </div>
  );
};

const FriendItem = () => {
  return (
    <div className={nav.friends_block_item}>
      <img
        className={nav.friends_block_img}
        src="https://cdn-icons-png.flaticon.com/512/4139/4139970.png"
        alt="icon"
      />
      <span>Andrei</span>
    </div>
  );
};

export default NavBlock;
