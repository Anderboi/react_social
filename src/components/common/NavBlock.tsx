import React from "react";
import { NavLink } from "react-router-dom";
import common from "../../Common.module.css";
import nav from "./NavBlock.module.css";

interface Props {
  isForMobile?: boolean | null;
  isAuth: boolean;
}

const NavBlock: React.FC<Props> = (props): JSX.Element => {
  
  const linkClass = (navData:any) =>
    `${navData.isActive ? nav.active : nav.nav_item} ${common.hover}`;

  return (
    <div
      className={`${
        !props.isForMobile ? nav.navBlock : nav.navBlock_mobile
      }`}
    >
      <nav className={nav.nav_menu}>
        <NavLink className={linkClass} to="/profile" data-testid="profile-link">
          Profile
        </NavLink>
        <NavLink
          className={linkClass}
          to="/messages"
          data-testid="messages-link"
        >
          Messages
        </NavLink>
         <NavLink className={linkClass} to="/news" data-testid="news-link">
          News
        </NavLink> 
         <NavLink className={linkClass} to="/music" data-testid="music-link">
          Music
        </NavLink> 
        <br></br>

        <NavLink className={linkClass} to="/Users" data-testid="users-link">
          Users
        </NavLink>
        <br></br>
        <NavLink
          className={linkClass}
          to="/settings"
          data-testid="settings-link"
        >
          Settings
        </NavLink>
        {/* <Outlet /> */}
      </nav>

      <div className={props.isAuth ? nav.nav_menu : common.display_none}>
        <h3>Friends</h3>
        <div className={nav.friends_block}>
          <FriendItem />
          <FriendItem />
          <FriendItem />
        </div>
      </div>
    </div>
  );
};

const FriendItem = (): JSX.Element => {
  return (
    <div className={nav.friends_block_item}>
      <img
        className={nav.friends_block_img}
        src="https://cdn-icons-png.flaticon.com/512/4139/4139970.png"
        alt="icon"
      />
      <span>Andrei</span>
    </div>
  );
};

export default NavBlock;
