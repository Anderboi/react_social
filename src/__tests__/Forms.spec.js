import React from "react";
import {create} from 'react-test-renderer'
import { TextForm } from '../components/common/Forms';

describe('Input Component', ()=>{
  test('Matches the snapshot', ()=>{
    const input = create(<TextForm/>);
    expect(input.toJSON()).toMatchSnapshot();
  })
})