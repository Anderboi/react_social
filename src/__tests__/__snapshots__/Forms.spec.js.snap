// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Matches the snapshot 1`] = `
<div>
  <input
    className="input"
  />
</div>
`;
